templates/al2/runtime/bootstrap.sh

```diff
@@ -46,6 +46,11 @@ function log {
 
 log "INFO: starting..."
 
+if [ "${EUID}" -ne 0 ]; then
+  log "ERROR: script must be run as root"
```

Comment:
for our usage this looks fine to me. In custom edge cases where this isn't satisfied i think it would just need a tweak to [`execute_command`](https://developer.hashicorp.com/packer/docs/provisioners/shell#execute-command-example)